runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -2233,21 +2442,25 @@ J9::ValuePropagation::doDelayedTransformations()
 
       const bool isLoad = callToTransform->isArrayElementLoadHelperCallTransform();
       const bool isStore = callToTransform->isArrayElementStoreHelperCallTransform();
+
       const bool isCompare = callToTransform->isObjectComparisonHelperCallTransform();
+      const bool isInlineVTCompare = callToTransform->_flags.testAny(ValueTypesHelperCallTransform::InlineVTCompare);
 
       const bool needsStoreCheck = callToTransform->_flags.testAny(ValueTypesHelperCallTransform::RequiresStoreCheck);
       const bool needsNullValueCheck = callToTransform->_flags.testAny(ValueTypesHelperCallTransform::RequiresNullValueCheck);
       const bool needsBoundCheck = callToTransform->_flags.testAny(ValueTypesHelperCallTransform::RequiresBoundCheck);
       const bool isFlattenedElement = callToTransform->_flags.testAny(ValueTypesHelperCallTransform::IsFlattenedElement);
 
-      // performTransformation was already checked for comparison non-helper call
-      // Only need to check for array element load or store helper calls
-      if (!isCompare && !performTransformation(
+      // performTransformation was already checked for non-VT comparison non-helper call.
+      // Need to check for array element load or store helper calls, or single field or multiple fields VT comparison
+      const bool hasCheckedNonVTComparePerformTransformation = isCompare ? (!isInlineVTCompare) : false;
```

Comment:
Maybe this is my personal preference for logical operations, but I think this might be easier to read as

```
const bool hasCheckedNonVTComparePerformTransformation = isCompare && !isInlineVTCompare;
```

Reply:
Addressed in 06a2cc2